frontend/src/modules/Form/components/popUp.jsx

```diff
@@ -0,0 +1,133 @@
+import React from "react";
+import "../components/Event.css";
+
+function EventModal({
+  isOpen,
+  toggleModal,
+  handleSubmit,
+  newEvent,
+  handleInputChange,
+  isEditMode,
+  addField,
+  removeField
+}) {
+  const modalClassName = isOpen ? "modal fade show" : "modal fade";
+
+  const onSubmit = (e) => {
+    e.preventDefault();
+    handleSubmit();
+  };
+
+  return (
+    <>
+      {isOpen && <div className="modal-backdrop fade show"></div>}
+      <div
+        className={modalClassName}
+        id="basicModal"
+        tabIndex="-1"
+      
+        style={{ display: isOpen ? "block" : "none",zIndex:99999 }}
+        aria-modal="true"
+        role="dialog"
+      >
+        <div className="modal-dialog" role="document">
+          <div className="modal-content">
+            <div className="modal-header">
+              <h5 className="modal-title" id="exampleModalLabel1">
+                {isEditMode ? "Update Form" : "Create Form"}
+              </h5>
+              <button
+                type="button"
+                className="btn-close"
+                onClick={toggleModal}
+                aria-label="Close"
+              ></button>
+            </div>
+            <form onSubmit={onSubmit}>
+              <div className="modal-body">
+                {!isEditMode && (
+                  <div className="mb-3">
+                    
+                    
+                  </div>
+                )}
+                <div className="mb-3">
+                  <label htmlFor="description" className="form-label">
+                    Description
+                  </label>
+                  <input
+                    type="text"
+                    id="description"
+                    className="form-control"
+                    placeholder="Enter Event Descrition"
+                    value={newEvent.description || ''}
+                    onChange={handleInputChange}
+                  />
+                </div>
+                <div className="mb-3">
+                  <label htmlFor="price" className="form-label">
+                    Price
+                  </label>
+                  <input
+                    type="text"
+                    id="price"
+                    className="form-control"
+                    placeholder="Enter Price"
+                    value={newEvent.price || ''}
+                    onChange={handleInputChange}
+                  />
+                </div>
+                {/* Dynamically generate input fields based on newEvent.data */}
+                {Object.keys(newEvent.data || {}).map((key) => (
```

Comment:
Check new event data is not undefined first 
```suggestion
          {    newEvent.data &&  Object.keys(newEvent.data || {}).map((key) => (
```